test: filter out picolib tests for ARC MWDT toolchain

evgeniy-paltsev opened the pull request:
Currently picolib isn't compatible with ARC MWDT toolchain, so don't try to build picolib tests in case of ARC MWDT toolchain usage.

galak reviewed (commented):
Is the CONFIG_PICOLIBC_SUPPORTED filter not sufficient?

stephanosio commented on tests/kernel/common/testcase.yaml:
```diff
@@ -29,6 +29,7 @@ tests:
       - CONFIG_CBPRINTF_FULL_INTEGRAL=y
   kernel.common.picolibc:
     filter: CONFIG_PICOLIBC_SUPPORTED
+    toolchain_exclude: arcmwdt
```
should modify `CONFIG_PICOLIBC_SUPPORTED` to be `n` when `arcmwdt` is selected.

keith-packard replied:
What is needed to make picolibc usable on arc mwdt toolchain?

keith-packard replied:
something like this?

[0001-libc-picolibc-Picolibc-is-not-supported-using-the-AR.txt](https://github.com/zephyrproject-rtos/zephyr/files/9303287/0001-libc-picolibc-Picolibc-is-not-supported-using-the-AR.txt)

evgeniy-paltsev replied:
Thanks @keith-packard !

Changed to proposed implementation.

pfalcon commented:
I recently reported a similar issue for gnuarmemb toolchain: https://github.com/zephyrproject-rtos/zephyr/issues/47275 . Picolibc should not be enabled for any toolchain but Zephyr SDK, as it's clearly not validated with anything else.

keith-packard commented:
> I recently reported a similar issue for gnuarmemb toolchain: #47275 . Picolibc should not be enabled for any toolchain but Zephyr SDK, as it's clearly not validated with anything else.

That's currently true in the Zephyr environment, but not true for upstream picolibc which is tested with the Zephyr SDK, Debian arm compiler, the GNU Arm Embedded Toolchain and clang. Getting Zephyr to understand how to use picolibc with those toolchains is a matter of Zephyr configuration, not picolibc validation.

pfalcon commented:
@keith-packard:

> That's currently true in the Zephyr environment, but not true for upstream picolibc which is tested with the Zephyr SDK, Debian arm compiler, the GNU Arm Embedded Toolchain and clang. Getting Zephyr to understand how to use picolibc with those toolchains is a matter of Zephyr configuration, not picolibc validation.

The talk in this context is strictly about Zephyr. I once again appreciate your efforts with Picolibc, but it's nice to have its further progress in a way that doesn't currently break anything in Zephy mainline (yes, by disabling picolibc in not proven configurations, while hopefully working on extending its coverage in dedicated branches). In that regard, I appreciate https://github.com/zephyrproject-rtos/zephyr/pull/48900, which fixed gnuarmemb issue I was seeing. Thanks!

cfriedt commented:
This looks ok to me - what's happening with the failing tests though?

keith-packard commented:
> This looks ok to me - what's happening with the failing tests though?

Agreed, it would be nice to actually fix the code so picolibc could build for zephyr with arc mwdt. Picolibc should build with any C compiler supporting C99 through C18.

evgeniy-paltsev committed:
teslibct: don't allow picolib tests for ARC MWDT toolchain

Currently picolib isn't compatible with ARC MWDT toolchain,
so don't try to build picolib tests in case of ARC MWDT toolchain
usage.

Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
Signed-off-by: Evgeniy Paltsev <PaltsevEvgeniy@gmail.com>

keith-packard reviewed (approved):
This looks good to me.
